airbyte-migration/src/main/resources/migrations/migrationV0_24_0/StandardSync.yaml

```diff
@@ -0,0 +1,64 @@
+---
+"$schema": http://json-schema.org/draft-07/schema#
+"$id": https://github.com/airbytehq/airbyte/blob/master/airbyte-config/models/src/main/resources/types/StandardSync.yaml
+title: StandardSync
+description: configuration required for sync for ALL taps
+type: object
+required:
+  - sourceId
+  - destinationId
+  - name
+  - catalog
+  - manual
+additionalProperties: false
+properties:
+  prefix:
+    description: Prefix that will be prepended to the name of each stream when it is written to the destination.
+    type: string
+  sourceId:
+    type: string
+    format: uuid
+  destinationId:
+    type: string
+    format: uuid
+  operationIds:
+    type: array
+    items:
+      type: string
+      format: uuid
+  connectionId:
+    type: string
+    format: uuid
+  name:
+    type: string
+  catalog:
+    existingJavaType: io.airbyte.protocol.models.ConfiguredAirbyteCatalog
+  status:
+    type: string
+    enum:
+      - active
+      - inactive
+      - deprecated
+  # Ideally schedule and manual should be a union, but java
+  # codegen does not handle the union type properly.
+  # When schedule is defined, manual should be false.
+  schedule:
```

Comment:
do you think something like that would be cleaner like:
```
schedule:
  type: object
  required:
    - type
  properties:
    type: enum(manual, frequency, other_type)
    frequency:
       type: object   
       properties:
          timeUnit:
             type: string
             enum:
               - minutes
               - hours
               - days
               - weeks
               - months
           units:
             type: integer
    other_type:
       type: object  
       properties:
          // blah 
```

This way you don't polute StandardSync with manual

Reply:
Ideally yes. The json-schema does support union with the `oneOf` keyword. However, this keyword is not supported by the java code generator `jsonschema2pojo`: https://github.com/joelittlejohn/jsonschema2pojo/issues/392

Reply:
yes, that's why I had to do what Michel is proposing for the `NotificationType` and `OperatorType` fields:
https://github.com/airbytehq/airbyte/blob/9e3bfee0863f6c23de45be9e0095a523818f84fc/airbyte-api/src/main/openapi/config.yaml#L1381

Reply:
Oh, I misunderstood what Michel was proposing. Basically we can make all those enum properties optional fields, and manually make sure only one of them exists on the object. That can work.

Reply:
Using the enum makes it a valid design (the best would be oneof ofcourse) instead of a hack